source/val/validation_state.cpp

```diff
@@ -394,5 +394,34 @@ void ValidationState_t::RegisterInstruction(
   if (id) {
     all_definitions_.insert(make_pair(id, &ordered_instructions_.back()));
   }
+
+  // If the instruction is using an OpTypeSampledImage as an operand, it should
+  // be recorded. The validator will ensure that all usages of an
+  // OpTypeSampledImage and its definition are in the same basic block.
+  for (auto i = 0; i < inst.num_operands; ++i) {
+    auto operand = inst.words[inst.operands[i].offset];
+    auto operand_inst = FindDef(operand);
+    if (operand_inst && SpvOpSampledImage == operand_inst->opcode()) {
+      RegisterSampledImageConsumer(operand, inst.result_id);
```

Comment:
Side note, not to be resolved by this issue.
The optimization framework has full definition-use tracking, of which this ID-tracking is a special case.  It's too bad the validator and optimizer don't share more code. :-(